paasta_tools/cli/cmds/spark_run.py

```diff
@@ -594,6 +594,21 @@ def run_docker_container(
     return 0
 
 
+def get_spark_app_name(original_docker_cmd):
+    # Use submitted batch name as default spark_run job name
+    if "spark-submit" in original_docker_cmd:
+        parser = argparse.ArgumentParser()
+        parser.add_argument("cmd", choices=["spark-submit"])
+        parser.add_argument("others", nargs="+")
+        args, _ = parser.parse_known_args(original_docker_cmd)
+        scripts = [arg for arg in args.others if arg.endswith(".py")]
+        if len(scripts) > 0:
+            batch_name = scripts[0].split("/")[-1].replace(".py", "")
```

Comment:
I'm a bit worried about other characters slipping through here.
You are assuming there is only a .py file. 

Do you know what the actual limitations of the spark_app_name are? If so, can you make a `sanitize_batch_name` function to do what happens on this line, but in a more robust way?

Reply:
Not really, the code will use the first xxx.py file, because spark-submit require the next argument to be the script we want to run, and then follow by the script's opts and args.  Therefore using the first one should be good enough. 

Also, the spark-submit doesn't recognize python module, and therefore we cannot do something like "spark-submit python -m module.path.to.script" to run the code. Which means we are always able to have a .py file in the argument.  The only exception is if people are running with script written in other language, and therefore is unable to find argument ends with .py, but In that case, we will simply return "paasta_spark_run" as what we have so far.

Reply:
It sounds like you covered the known knowns.
In general I'm always very defensive against user-inputed filenames. Eventually this code will have to handle emoji, non-ascii characters, and uncode left-to-right characters! (j/k maybe)

Reply:
one thing I noticed is that the run_docker_container will use the spark app name as its `--name` , but docker container does  require name to be [a-zA-Z0-9][a-zA-Z0-9_.-]. Therefore I made a separate sanitize_container_name, so that it won't break the docker run.  Wondering if it is a better idea to do the sanitize in the spark app instead.  The spark doesn't seem to limit anything on the naming, and therefore it is not going to cause problem on the spark side.